packages/integrations/node/src/standalone.ts

```diff
@@ -1,75 +1,90 @@
-import type { NodeApp } from 'astro/app/node';
+import http from 'node:http';
 import https from 'https';
-import path from 'node:path';
-import { fileURLToPath } from 'node:url';
-import { getNetworkAddress } from './get-network-address.js';
-import { createServer } from './http-server.js';
-import middleware from './nodeMiddleware.js';
+import fs from 'node:fs';
+import enableDestroy from 'server-destroy';
+import createAppListener from './serve-app.js';
+import createStaticListener from './serve-static.js';
+import logListeningOn from './log-listening-on.js';
+import type { NodeApp } from 'astro/app/node';
 import type { Options } from './types.js';
+import type { PreviewServer } from 'astro';
 
-function resolvePaths(options: Options) {
-	const clientURLRaw = new URL(options.client);
-	const serverURLRaw = new URL(options.server);
-	const rel = path.relative(fileURLToPath(serverURLRaw), fileURLToPath(clientURLRaw));
-
-	const serverEntryURL = new URL(import.meta.url);
-	const clientURL = new URL(appendForwardSlash(rel), serverEntryURL);
-
+export default function standalone(app: NodeApp, options: Options) {
+	const port = process.env.PORT ? Number(process.env.PORT) : options.port ?? 8080;
+	// Allow to provide host value at runtime
+	const host_ = typeof options.host === "boolean" ? "127.0.0.1" : options.host
+	const host = process.env.HOST ?? host_;
```

Comment:
Naming adjustment for clarity

```suggestion
	const hostOptions = typeof options.host === "boolean" ? "127.0.0.1" : options.host
	const host = process.env.HOST ?? hostOptions;
```